nilearn/datasets.py

```diff
@@ -2497,22 +2498,24 @@ def fetch_abide_pcp(data_dir=None, n_subjects=None, pipeline='cpac',
     # field. This can be
     # done simply with pandas but we don't want such dependency ATM
     # pheno = pandas.read_csv(path_csv).to_records()
-    pheno_f = open(path_csv, 'r')
-    pheno = ['i' + pheno_f.readline()]
-    # This regexp replaces commas between double quotes
-    for line in pheno_f:
-        pheno.append(re.sub(r',(?=[^"]*"(?:[^"]*"[^"]*")*[^"]*$)', ";", line))
-    pheno_f.close()
-    pheno = '\n'.join(pheno)
-    pheno = StringIO.StringIO(pheno)
+    with open(path_csv, 'r') as pheno_f:
+        pheno = ['i' + pheno_f.readline()]
+
+        # This regexp replaces commas between double quotes
+        for line in pheno_f:
+            pheno.append(re.sub(r',(?=[^"]*"(?:[^"]*"[^"]*")*[^"]*$)', ";", line))
+
+    # bytes (encode()) needed for python 2/3 compat with numpy
+    pheno = '\n'.join(pheno).encode()
+    pheno = BytesIO(pheno)
     # We enforce empty comments because it is 'sharp' by default
-    pheno = np.recfromcsv(pheno, comments=[], case_sensitive=True)
+    pheno = np.recfromcsv(pheno, comments='xyzpdq', case_sensitive=True)
```

Comment:
... fixed all, except this one.  I am investigating this, I found this to be necessary (but will double-check).